File: AbetOrder/OpenDataAccess/OpenDataAccess.Data/DataAccess.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Configuration;

namespace OpenDataAccess.Data
{
    public static class DataAccess
    {
        private static string _ConnectionString { get; set; }
        private static string _ServerClient { get; set; }

        public static T GetDataBase<T>() where T : IDataBase
        {
            IDataBase dataBase = null;
            if (typeof(T) == typeof(ISqlDataBase))
            {
                dataBase = new SqlDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else if (typeof(T) == typeof(IOracleDataBase))
            {
                dataBase = new OracleDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else if (typeof(T) == typeof(IMySqlDataBase))
            {
                dataBase = new MySqlDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else
            {
                throw new ArgumentException();
            }
            return (T)dataBase;
        }

        public static IDataBase GetDataBase()
        {
            IDataBase dataBase = null;
            if (ServerClient.Equals("Sql"))
            {
                dataBase = new SqlDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else if (ServerClient.Equals("Oracle"))
            {
                dataBase = new OracleDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else if (ServerClient.Equals("MySql"))
            {
                dataBase = new MySqlDataBase();
                dataBase.ConnectionString = ConnectionString;
            }
            else
            {
                throw new ArgumentException();
            }

            return dataBase;
        }

        private static string ServerClient
        {
            get
            {
                if (string.IsNullOrEmpty(_ServerClient))
                {
                    _ServerClient = ConfigurationManager.AppSettings["ServerClient"];
                    _ServerClient = string.IsNullOrEmpty(_ServerClient) ? "Sql" : _ServerClient;
                }
                return _ServerClient;
            }
        }

        private static string ConnectionString
        {
            get
            {
                if (string.IsNullOrEmpty(_ConnectionString) && ConfigurationManager.ConnectionStrings["ConnectionString"] != null)
                {
                    _ConnectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
                }
                return _ConnectionString;
            }
        }
    }
}

File: JavaWebFramework/src/OpenDataAccess/Data/DataAccess.java
package OpenDataAccess.Data;

import OpenDataAccess.Utility.AppSettings;

public class DataAccess {
    private static String _ConnectionString = null;
    private static String _ServerClient = null;
    private static String _User = null;
    private static String _Password = null;

    public static <T extends IDataBase> T GetDataBase(Class<T> cls) {
        IDataBase dataBase = null;
        if (cls.getClass().equals(ISqlDataBase.class)) {
            dataBase = new SqlDataBase(GetConnectionString(), GetUser(),GetPassword());
        } else if (cls.getClass().equals(IOracleDataBase.class)) {
            dataBase = new SqlDataBase(GetConnectionString(), GetUser(),GetPassword());
        } else if (cls.getClass().equals(IMySqlDataBase.class)) {
            dataBase = new MySqlDataBase(GetConnectionString());
        } else {
            throw new IllegalArgumentException();
        }
        return (T) dataBase;
    }

    public static IDataBase GetDataBase() {
        IDataBase dataBase = null;
        String client = GetServerClient();
        if (client.equals("Sql")) {
            dataBase = new SqlDataBase(GetConnectionString(), GetUser(),GetPassword());
        } else if (client.equals("Oracle")) {
            dataBase = new OracleDataBase(GetConnectionString(), GetUser(),GetPassword());
        } else if (client.equals("MySql")) {
            dataBase = new MySqlDataBase(GetConnectionString());
        } else {
            throw new IllegalArgumentException();
        }

        return dataBase;
    }

    private static String GetServerClient() {
        if (OpenDataAccess.Utility.Common.IsNullOrEmpty(_ServerClient)) {
            _ServerClient = AppSettings.ServerClient;
            _ServerClient = OpenDataAccess.Utility.Common.IsNullOrEmpty(_ServerClient) ? "Sql" : _ServerClient;
        }
        return _ServerClient;
    }

    private static String GetConnectionString() {
        if (OpenDataAccess.Utility.Common.IsNullOrEmpty(_ConnectionString))
            _ConnectionString = AppSettings.ConnectionString;
        return _ConnectionString;
    }

    private static String GetUser() {
        if (OpenDataAccess.Utility.Common.IsNullOrEmpty(_User)) _User = AppSettings.DbUser;
        return _User;
    }

    private static String GetPassword() {
        if (OpenDataAccess.Utility.Common.IsNullOrEmpty(_Password)) _Password = AppSettings.DbPassword;
        return _Password;
    }
}
